src/viadot/sources/eurostat.py

```diff
@@ -0,0 +1,290 @@
+            TypeError: If self.requested_columns is different type than a list containing strings.
+        Returns:
+            pd.DataFrame: Final DataFrame or raise prefect.logger.error, if issues occur.
+        """
+
+        self.dataset_code = dataset_code
+        self.params = params
+        if not isinstance(self.params, dict) and self.params is not None:
+            raise TypeError("Params should be a dictionary.")
+
+        self.base_url = f"{base_url}{self.dataset_code}?format=JSON&lang=EN"
+        self.requested_columns = requested_columns
+        if not isinstance(requested_columns, list) and requested_columns is not None:
+            raise TypeError("Requested columns should be provided as list of strings.")
+
+        # getting response from API
+        try:
+            response = handle_api_response(self.base_url, params=self.params)
+            data = response.json()
+            data_frame = self.eurostat_dictionary_to_df(["geo", "time"], data)
+        except APIError:
+            self.make_params_validation()
+
+        if data_frame.empty:
+            try:
+                self.make_params_validation()
+                self._handle_if_empty(if_empty)
+            except SKIP:
+                return pd.DataFrame()
+
+        # merging data_frame with label and last updated date
+        label_col = pd.Series(str(data["label"]), index=data_frame.index, name="label")
+        last_updated__col = pd.Series(
+            str(data["updated"]),
+            index=data_frame.index,
+            name="updated",
+        )
+        data_frame = pd.concat([data_frame, label_col, last_updated__col], axis=1)
+
+        # requested columns validation
+        if self.requested_columns is None:
+            return data_frame
+        else:
+            columns_list = data_frame.columns.tolist()
+            columns_list = [str(column).casefold() for column in columns_list]
+            needed_column_after_validation = []
+            non_available_columns = []
+
+            for column in self.requested_columns:
+                # Checking if user column is in our dataframe column list
+                column = str(column).casefold()
+
+                if column in columns_list:
+                    needed_column_after_validation.append(column)
+                else:
+                    non_available_columns.append(column)
+
+            # Error logger
+            if non_available_columns:
+                self.logger.error(
+                    f"Name of the columns: '{' | '.join(non_available_columns)}' are not in DataFrame. Please check spelling!\n"
+                    f"Available columns: {' | '.join(columns_list)}"
+                )
+                raise ValueError("Provided columns are not available!")
+
+            data_frame = data_frame.loc[:, needed_column_after_validation]
+
+            # Additional validation from utils
+            validate(df=data_frame, tests=tests)
+
+            return data_frame
```

Comment:
1) Would be good to put this code in some helper method because this `to_df()` is already pretty long (~110 lines; usually it's best to stay within ~50 (you should see the entire function without scrolling)).

2) Why do you only run `validate()` if `self.requested_columns` is specified, shouldn't it be executed in either case?

Reply:
1. I put requested_columns logic into method: requested_columns_validation().
2. I belive it is not, as validate() is outside if statement of requested_columns:
        # Validation and transformation of requested column
        if self.columns is not None:
            self.requested_columns_validation(data_frame=data_frame)

        # Additional validation from utils
        validate(df=data_frame, tests=self.tests)

Reply:
AD 2) Sorry, this is unreadable. For the future, you can paste code samples using the code block Markdown syntax: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting

But I checked the code and see you have fixed the logic: you're now running `validate()` regardless of whether you're filtering any columns or not (here you were only validating if you were filtering out some columns).